js/main.js

```diff
@@ -0,0 +1,66 @@
+var $globeButton = document.querySelector('.globe-button');
+var $globeContainer = document.querySelector('.globe-container');
+var $atlasButton = document.querySelector('.header-button');
+var $countryGenerator = document.querySelector('.generator-container');
+var $countryFlag = document.querySelector('.country-flag');
+var $countryName = document.querySelector('.country-name');
+var $nextButton = document.querySelector('.next-button');
+
+function globeButtonEvent(event) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://restcountries.com/v3.1/all');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var countries = xhr.response;
+    var randomIndex = Math.floor(Math.random() * countries.length);
+    $countryFlag.setAttribute('src', countries[randomIndex].flags.png);
+    $countryName.innerHTML = countries[randomIndex].name.common;
+  });
+  xhr.send();
+  viewSwap('country-generator');
+  data.view = 'globe';
+}
+$globeButton.addEventListener('click', globeButtonEvent);
+
+function atlasButtonEvent(event) {
+  viewSwap('globe');
+  data.view = 'globe';
+}
+
+$atlasButton.addEventListener('click', atlasButtonEvent);
+
+function nextButtonEvent(event) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', 'https://restcountries.com/v3.1/all');
+  xhr.responseType = 'json';
+  xhr.addEventListener('load', function () {
+    var countries = xhr.response;
+    var randomIndex = Math.floor(Math.random() * countries.length);
+    $countryFlag.setAttribute('src', countries[randomIndex].flags.png);
+    $countryName.innerHTML = countries[randomIndex].name.common;
+  });
```

Comment:
Instead of sending a request to get countries every time the user asks for another, you should save the list from your first request in `globeButtonEvent` into your data model, and use that as your source from that point on.